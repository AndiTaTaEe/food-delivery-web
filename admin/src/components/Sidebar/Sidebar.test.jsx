import {describe, it, expect, beforeEach} from "vitest";
import {render, screen} from "@testing-library/react";
import {BrowserRouter} from "react-router-dom";
import Siderbar from "./Siderbar";
import "@testing-library/jest-dom";
import {assets} from "../../assets/assets";

vi.mock('../../assets/assets', () => ({
    assets: {
        add_icon: '/mock-path/add_icon.png',
        list_icon: '/mock-path/order_icon.png',
        order_icon: '/mock-path/order_icon.png'
    }
}));

describe ("Sidebar component", () => {
    beforeEach(() => {
        render(
            <BrowserRouter>
                <Siderbar />
            </BrowserRouter>
        );
    });


it("renders sidebar component", () => {
    const sidebarElement = screen.getByTestId("mock-sidebar");
    expect(sidebarElement).toBeInTheDocument();
    expect(sidebarElement).toHaveClass('sidebar');
});

it("displays nav links", () => {
    const addItemsLink = screen.getByText(/Add Items/i);
    const listItemsLink = screen.getByText(/List Items/i);
    const ordersLink = screen.getByText(/Orders/i);

    expect(addItemsLink).toBeInTheDocument();
    expect(listItemsLink).toBeInTheDocument();
    expect(ordersLink).toBeInTheDocument();
});

it("sidebar has correct nav paths", () => {
    const addItemsLink = screen.getByText(/Add Items/i);
    const listItemsLink = screen.getByText(/List Items/i);
    const ordersLink = screen.getByText(/Orders/i);

    expect(addItemsLink.closest('a')).toHaveAttribute('href', '/add');
    expect(listItemsLink.closest('a')).toHaveAttribute('href', '/list');
    expect(ordersLink.closest('a')).toHaveAttribute('href', '/orders');
});

});

describe("Snapshot testing", () => {
    it("matches the snapshot", () => {
        const {container} = render(
            <BrowserRouter>
                <Siderbar />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    })
})