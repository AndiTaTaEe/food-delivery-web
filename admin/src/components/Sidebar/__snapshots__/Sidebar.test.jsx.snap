// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot testing > matches the snapshot 1`] = `
<div>
  <div
    class="sidebar"
    data-testid="mock-sidebar"
  >
    <div
      class="sidebar-options"
    >
      <a
        class="sidebar-option"
        data-discover="true"
        href="/add"
      >
        <img
          alt=""
          src="/mock-path/add_icon.png"
        />
        <p>
          Add Items
        </p>
      </a>
      <a
        class="sidebar-option"
        data-discover="true"
        href="/list"
      >
        <img
          alt=""
          src="/mock-path/order_icon.png"
        />
        <p>
          List Items
        </p>
      </a>
      <a
        class="sidebar-option"
        data-discover="true"
        href="/orders"
      >
        <img
          alt=""
          src="/mock-path/order_icon.png"
        />
        <p>
          Orders
        </p>
      </a>
    </div>
  </div>
</div>
`;
